DeviceSDK/src/main/java/org/astarteplatform/devicesdk/AstarteDeviceIdUtils.java

```diff
@@ -0,0 +1,36 @@
+package org.astarteplatform.devicesdk;
+
+import java.nio.ByteBuffer;
+import java.util.Base64;
+import java.util.UUID;
+import org.astarteplatform.devicesdk.util.UUID5;
+
+/** The class helps to generate deviceId for Astarte Device API */
+public class AstarteDeviceIdUtils {
+
+  /**
+   * Generate a device Id in a random way.
+   *
+   * @return the device Id generated
+   */
+  public static String generateId() {
+    UUID randomUUID = UUID.randomUUID();
+    Base64.Encoder encoder = Base64.getUrlEncoder().withoutPadding();
+    return encoder.encodeToString(randomUUID.toString().getBytes());
+  }
+
+  /**
+   * Generate a device Id based on UUID namespace identifier and a uniqueData.
+   *
+   * @return the device Id generated
+   */
+  public static String generateId(UUID namespace, String uniqueData) {
```

Comment:
I think we should provide our users an usage example, such as with a namespace and a MAC address string.

Reply:
Yes, I will provide a full usage example and README in another PR.